Fix property binding issues to make Koog Spring Boot auto-configuration work and add tests

dmitrysulman opened the pull request:
Unfortunately, the recently introduced [Koog Spring Boot Starter](https://docs.koog.ai/spring-boot/) is currently unusable due to fundamental issues with configuration property binding. The `koog-spring-boot-starter` module contains no tests, which allowed these bugs to go unnoticed. There was even an attempt to [add tests](https://github.com/JetBrains/koog/pull/424), but that PR was closed - likely because the tests failed (as expected, given the issues).

This PR fixes the binding problems and introduces tests to prevent regressions. Below I explain the issues in detail, with references to the relevant Spring Boot documentation.

### Incorrect field names for nested properties
The starter uses the [`KoogProperties`](https://github.com/JetBrains/koog/blob/9714223f2bce944166b3d128099980afc160c6a1/koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/KoogProperties.kt) class with the `@ConfigurationProperties(prefix = "ai.koog")` annotation to bind configuration properties. Providers such as OpenAI, Anthropic, and Google are represented by nested `ProviderKoogProperties` classes.

Example OpenAI configuration:
```
ai.koog.openai.api-key=${OPENAI_API_KEY}
ai.koog.openai.base-url=https://api.openai.com
```

In the code, OpenAI properties were mapped to a field named `openAIClientProperties`:
https://github.com/JetBrains/koog/blob/9714223f2bce944166b3d128099980afc160c6a1/koog-spring-boot-starter/src/main/kotlin/ai/koog/spring/KoogProperties.kt#L92-L94

However, Spring Boot relies purely on Java/Kotlin field names during binding - it cannot infer that `openai` should map to `openAIClientProperties`. According to the [Spring Boot configuration property binding rules](https://docs.spring.io/spring-boot/reference/features/external-config.html#features.external-config.typesafe-configuration-properties.java-bean-binding), the field must be named exactly `openai`:
```kotlin
@NestedConfigurationProperty
public var openai: ProviderKoogProperties =
    ProviderKoogProperties(baseUrl = "https://api.openai.com")
```

This mismatch prevented all nested properties from binding correctly.

### Solution
1. Introduce correct separate configuration properties classes for each LLM provider.
2. Add `KoogAutoConfigurationTest` to validate auto-configuration and property binding.
3. Clean up `build.gradle.kts` by removing unused `compileOnly` dependencies.

### Result
With these fixes, the Koog Spring Boot Starter now binds configuration properties correctly and is covered by automated tests to prevent future regressions.

Tagging everyone who took part in [the initial Spring Boot starter PR](https://github.com/JetBrains/koog/pull/334): @kpavlov @skarpovdev @devcrocod @aozherelyeva

kpavlov commented:
Thank you, @dmitrysulman for your PR and especially for the tests. I think we could go even further and extract provider properties from `KoogProperties` into separate classes to make the code more modular and [avoid coupling](https://en.m.wikipedia.org/wiki/Single-responsibility_principle). In this case we don't even need the `@NestedConfigurationProperties` annotation. In auto-configuration class we should use [`@ConditionallyOnBean`](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/autoconfigure/condition/ConditionalOnBean.html).

Regarding changing `val` to `var`: normally, Spring boot supports `val` properties and I would prefer keeping it Properties inmutable.

dmitrysulman commented:
@kpavlov thank you for the feedback!

I’ve updated the PR and introduced separate configuration properties classes for each LLM provider, as you suggested.

I'm not fully sure what you meant by this part:

> In auto-configuration class we should use [@ConditionallyOnBean](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/autoconfigure/condition/ConditionalOnBean.html).

The `@ConditionalOnBean` annotation is used to enable auto-configuration only when specific bean(s) are already present in the application context. Could you clarify which bean(s) you expect to be used here?

kpavlov commented:
Thank you, @dmitrysulman! Looks good and Property beans are immutable 👍🏻 

> I'm not fully sure what you meant by this part:
> 
> > In auto-configuration class we should use [@ConditionallyOnBean](https://docs.spring.io/spring-boot/api/java/org/springframework/boot/autoconfigure/condition/ConditionalOnBean.html).
> 
> The `@ConditionalOnBean` annotation is used to enable auto-configuration only when specific bean(s) are already present in the application context. Could you clarify which bean(s) you expect to be used here?

You are right, it might be better to use something else for autoconfiguration, e.g. configure LLM cllient bean only when required API key is present in the configuraiton. But let's take it out of scope of this PR.

kpavlov commented:
@dmitrysulman, `@NestedConfigurationProperty` is no longer needed. Let's clean it up and merge ;)

kpavlov commented on koog-spring-boot-starter/src/test/kotlin/ai/koog/spring/KoogAutoConfigurationTest.kt:
```diff
@@ -0,0 +1,282 @@
+            }
+    }
+
+    private inline fun <reified T> getPrivateFieldValue(instance: T, fieldName: String): Any? {
```
Let's address it in a separate PR. It make sense to make some fields public, but still immutible. For visibility
